### PR TITLE
Gives gangs new surplus rifle, mounted machine gun, and more

### PR DESCRIPTION
**Pull request by Robustin**
We're pretty overdue for a gun that isn't design to kill you in under 2 seconds. The new surplus rifle (8 points) requires two hands to fire, is very bulky, and it take a few seconds between each shot. Its a cheap way for people to defend a gang's turf without giving them a murderbone tool. Uses 10mm rounds (30 damage a shot). 

The mounted machine gun (50 points) is something I've had sitting around. I wanted to make gang gunfights a little more tactical which is why I added stuff like the sniper rifle. The mounted machine gun is something you buckle into and makes both hands unusable while using. It fires a volley of 10 rounds of 20 damage bullets with moderate spread over 2 seconds, then requires 2.5 seconds to "cool off" before another volley can be fired. The machine gun never runs out of ammo and extends sight range by 3, so its an excellent suppression weapon, but its spread and low damage per bullet means you aren't going to be blowing people away either. Its stationary status mean that the gun is almost useless in open spaces (if you can flank the gunner after their initial volley you have almost 5 seconds to put them down before they can fire again) or any position where someone can fire on you from cover.

I'm also giving gangs access to a sawn-off improvised shotgun (6 points) that fires only one shell at a time.

This PR also adds my previously added gear to the item list I overlooked, which should allow them to be purchased now. 

🆑 Robustin
add: Gang influence is now decentralized, each gangster has their own influence that they can increase by spraying (and protecting) their tags and new influence-enhancing bling. 
add: Gang uniforms are now created based on your gang's color and can be purchased by any gang member. They will increase the wearer's influence and provide protection, but will make it fairly obvious which gang you belong to. 
add: Gangs have access to a new surplus rifle; it is a semi-automatic rifle that is very bulky and has a very low rate of fire. Gang members can buy this gun for just 8 influence.
add: Gangs have access to the new machine gun turret; it unleashes a volley of bullets with an extended view range. It does not run out of ammo, but a significant delay between volleys and its stationary nature leaves the gunner vulnerable to flanking and return fire. Holding down the trigger will allow you to aim the gun while firing. It will cost gangs 50 influence. 
add: The sawn-off improvised shotgun is now available to gangs for 6 influence. With buckshot shells being easy to produce or purchase, this gun gives the most "bang for your buck" but its single-shell capacity will leave you vulnerable during a pitched gunfight. 
add: The Wetwork boots give gangs access to a lightly armored noslip variant. 
tweak: The armored gang outfits are now slightly more resistant to ballistic and bomb damage
/🆑

### CODE REVIEW

**Comment by coiax** [comment]
> doesn't run out of ammo

Oh boy.

**Comment by FantasticFwoosh** [comment]
Imminent wall of death coming when  people build a solid wall of these, it'll just be 3 gangs sitting on different ends of the station in a stalemate because they've built a tower defence course.

> I can atleast count on 1 shitter camping the arrivals shuttle with it.

**Comment by ma44** [comment]
Throw a grenade or gas, something like clf3 + smoke is most likely to kill anything that decides to stand still.

**Comment by Robustin** [comment]
I believe its underpowered right now but you guys can prove me wrong.

MG = 50TC
Sniper rifle = 40TC

Since the gunner can't manipulate the cooldown, you can wait between volleys and easily dunk the gunner with any ranged weapons. The gunner can't turn the gun after it starts firing either so it someone is blitzing your position and runs around your cone of fire, you pretty much have to abandon ship.

Also anyone who gets rid of the gunner can easily steal these.

**Comment by Repukan** [comment]
>Adding guns when RnD can just print them or cargo can order
>Adding stationary defenses when toxins exist

File: code/game/machinery/porta_turret/portable_turret.dm
Comment (Cyberboss) [inline code/game/machinery/porta_turret/portable_turret.dm]: Is this not set by the base proc?

File: code/game/machinery/porta_turret/portable_turret.dm
Comment (Cyberboss) [inline code/game/machinery/porta_turret/portable_turret.dm]: Use reset_perspective

File: code/game/machinery/porta_turret/portable_turret.dm
Comment (Cyberboss) [inline code/game/machinery/porta_turret/portable_turret.dm]: forceMove

File: code/game/machinery/porta_turret/portable_turret.dm
Comment (Cyberboss) [inline code/game/machinery/porta_turret/portable_turret.dm]: I swear theres a proc for this

File: code/game/machinery/porta_turret/portable_turret.dm
Comment (Cyberboss) [inline code/game/machinery/porta_turret/portable_turret.dm]: Use the define for whatever layer 4 is and the +/- 0.1 to it

**Comment by Iamgoofball** [comment]
>tachanka turret

when do gangs get fuze charges too?

File: code/game/machinery/porta_turret/portable_turret.dm
Comment (Cyberboss) [inline code/game/machinery/porta_turret/portable_turret.dm]: remove .loc

File: code/game/machinery/porta_turret/portable_turret.dm
Comment (Cyberboss) [inline code/game/machinery/porta_turret/portable_turret.dm]: remove .loc

**Review by Cyberboss — CHANGES_REQUESTED** [review]

**Comment by Robustin** [comment]
@Repukan 

Those things will be changed very soon - and even then its a stupid argument to say "well we shouldn't give people guns because cargo can get cheap guns" when the whole idea is to not make cargo the exclusive gun store for the station.

**Comment by FantasticFwoosh** [comment]
Its probably put down to the factor that ballistic guns are generally more reliable & damaging (everyone grabs combat shotties probably more in favour than grabbing laser rifles)

Gang mode guns can be devastating en-mass because of bullets, lasers run out & and need to recharge while guns just keep going with ammo to spare. And with gang points in the tank to just buy more/virtually unlimited ammo its strong.

**Comment by Robustin** [comment]
Gang is losing access to the armory and cargo guns, they need something to fill the gap and I'm adding guns that aren't just mass-murder devices. The sniper, sawn-off, surplus, MG, etc. are all really bad for just running down the hall and killing anything that moves.

This helps add depth so that defenses aren't just wallspam and uzis.

**Comment by Ausops** [comment]
if we allow gang we can try this

**Comment by kevinz000** [comment]
why do i get yelled at for gunbloat when robustin gets all the fun ;-;

**Comment by Bawhoppen** [comment]
gunbloat isn't a real thing don't listen to those people

**Comment by Robustin** [comment]
Nothing is gunbloat if it fills a significant role in ranged combat that isn't filled by anything else.

Low DPS non-stunning ranged weapon? We literally have none. 

Weapon focused on suppressing chokepoints and enemy groups instead of outright killing them? Zip.

**Comment by BeeSting12** [comment]
all hail lord tachanka

**Comment by lzimann** [comment]
@tgstation/commit-access

**Comment by Steelpoint** [comment]
Seems fair, gang get access to firearms but they are not OP monster weapons that have the most invisible gun sprite in the game, can hide in your backpack, and kill someone in under two seconds.

I'm also interested in the idea of cutting off gangs access to cargo and armoury guns, look forward to seeing Robustin's further work on gang.

File: code/game/machinery/porta_turret/portable_turret.dm
Comment (Robustin) [inline code/game/machinery/porta_turret/portable_turret.dm]: Didn't remove zoom, changing it back

**Comment by Robustin** [comment]
I overhauled the machine gun to be a lot more responsive than just "vomits bullets at a location for a couple seconds then cools down".

Thanks to being egged on by a certain maintainer I have made the machine gun (and its new admin only counterpart) into autoclicker weapons with a behavior revamp. What does this mean? The weapon will now begin firing upon you depressing the mouse button and it will track your mouse's movement as you depress the mouse. As long as the gun is firing it will constantly be updating to your mouse's location. The weapon's spread prevents this from being too OP. I also reduced the gun's integrity to 100 from the default 200 so that it can more easily be dispatched by hostile gunfire. While the gunfire window is now doubled to around 4 seconds, you also now have a 5 second window to attack the gun while it cools off.

**Comment by Tacolizard** [comment]
>Low DPS non-stunning ranged weapon? We literally have none.

hmm my pipe pistol would've filled that niche huh imagine that

**Comment by Robustin** [comment]
I think your pipe pistol was doing fine but coding and drama derailed it.

**Comment by Robustin** [comment]
How the fuck do I have 2 turret icon conflicts in 2 days.

**Comment by Dorsisdwarf** [comment]
because i changed turrets.dmi twice in two days

File: code/game/gamemodes/gang/gang_items.dm
Comment (ChangelingRain) [inline code/game/gamemodes/gang/gang_items.dm]: </span>

**Comment by kevinz000** [comment]
wtf i like robustin now?

**Comment by Steelpoint** [comment]
Having a lot of low damage, low rate of fire, weapons would make combat a more drawn out affair which I think would be of benefit.

It would also make the Uzi (if you decide to keep it) far more valuable but also something rare to see. Considering one Uzi is worth six rifles you can argue that six guys with crappy guns is better than one guy with a good gun.

**Comment by Robustin** [comment]
@Steelpoint I agree, I'm excited about the opportunity to have a mode that doesn't have much in the way of reliable stuns (the biggest thing I'm worried about once sec is gone is the power of telescopic batons). As far as gun balance I agree that encouraging "ambush" gameplay where the focus is destroying your opponent before they can react is bad, but its not the Uzi that I'm primarily worried about.

I have mixed feelings about the Uzi. I think its the best all-around gun for the Gang but I don't think its that strong. 

Its 32 rounds with a 2-shot burst that does 20 damage per shot. The gun costs 60 influence and each mag costs 40 influence.

Meanwhile the stechkin, I increased the cost to 30 from 25 but that's still 50% of the uzi's cost and it does 30 damage per shot. I fear the stechkin becoming everyone's default weapon and I think the DPS is high enough to allow you to flatten most people before they realize you've got a couple black pixels in your hand. I'm considering increasing the ballistic armor on gang outfits but I'm also thinking about changing the stechkin to a "Gang variant" that is has a more noticeable icon, or adding more ballistic armor to the gang outfit so that gunfights aren't about "kill your opponent in under a sec".

I'm leaning toward the latter. I don't think a guy with an Uzi could even hope to take on 6 guys with surplus rifles. I would honestly feel comfortable taking on an Uzi user solo with a surplus if I could keep my distance and take cover between shots. If the game proves otherwise then sure I'll increase the price more.

**Comment by Steelpoint** [comment]
I do think the Surplus Rifle is a strong enough weapon, it won't hold up long against guys with Lasers and Riot Shields but since sec is MIA in this mode that is no issue.

If anything I suggest just removing the UZI or allowing one to be in existence at any time for each gang.

**Comment by Steelpoint** [comment]
Perhaps look to changing the Sketchin to a different name but make its MOB sprite very large and obvious.

**Comment by Steelpoint** [comment]
Now that I think about it, I think we should remove the higher tier weapons above the surplus rifle. Keep it to the surplus rifle, one shell shotgun and the machine gun. 

The reason I say this is due to your other proposed change of making influence accumulate on an individual basis, not a group basis. If I accumulate influence individually I'm more incentivised to save my influence to buy a Sketchin or a UZI.

**Comment by Robustin** [comment]
Since Gang War is entering a very experimental state I'd rather keep the list as open as possible and then shave it down based on real experiences and/or data than start small and then fight people on hypotheticals.

**Comment by AdamElTablawy** [comment]
Could you not theoretically just place a bunch of these in a hallway/around a corner and make a wall of death? 

Remember that shotgun buckshot had 'spread' too but it was pretty good at hurting like shit even at long range, and that's only a few pellets that do low damage.

This is a 20-bullet stream. While the shotgun is portable and not destroyable like this is, spread means little in this sort of scenario.

**Comment by Bawhoppen** [comment]
@Adam some shotgun pellets always go forwards, the spread for shotguns isn't that random

**Comment by Bawhoppen** [comment]
Also sorry for pinging you random guy

**Comment by Robustin** [comment]
If you place them anywhere obvious someone can just return fire down the hall. The guns only have 100hp (most machines have 200) so you cant really afford to put them somewhere in the open. I would imagine that if 4 stechkin rounds trash your 50 influence machine gun, that you would avoid making "walls of death" in the future.

But who knows. My experience is that standing still is a death sentence in this game, a single cooked grenade could killed 3 gunners and 3 guns. Also with the gang uniform each MG bullet does ~13 damage, each tile will get roughly 33% of the gunfire from the MG, so any given tile is looking at ~40 damage per second from a machine gun at range - which is still less than someone just plugging you with a stechkin.

**Comment by lzimann** [comment]
@tgstation/commit-access This is just adding, once again, more traitor gear to gang. I'll close this in 24h if no one merges.

**Comment by Steelpoint** [comment]
None of this is traitor gear, its all unique and all worse than the standard of gear Traitors get.

**Comment by lzimann** [comment]
Doesn't matter, I still dislike.

File: code/game/machinery/porta_turret/portable_turret.dm
Comment (RemieRichards) [inline code/game/machinery/porta_turret/portable_turret.dm]: ```reset_perspective()``` is the wrong proc for this, that's to do with client eye positioning.
```change_view()``` is correct.

File: code/game/machinery/porta_turret/portable_turret.dm
Comment (RemieRichards) [inline code/game/machinery/porta_turret/portable_turret.dm]: Defines please, for example, I'm sure 4 is like, ```OBJ_LAYER``` or something, so you'd do ```layer = OBJ_LAYER + 0.2``` so people are aware it's "0.2 layers above the Object layer"

File: code/game/machinery/porta_turret/portable_turret.dm
Comment (RemieRichards) [inline code/game/machinery/porta_turret/portable_turret.dm]: same again

File: code/game/machinery/porta_turret/portable_turret.dm
Comment (RemieRichards) [inline code/game/machinery/porta_turret/portable_turret.dm]: See, you did it here!

File: code/game/machinery/porta_turret/portable_turret.dm
Comment (RemieRichards) [inline code/game/machinery/porta_turret/portable_turret.dm]: no ```var/``` in proc arguments

File: code/game/machinery/porta_turret/portable_turret.dm
Comment (RemieRichards) [inline code/game/machinery/porta_turret/portable_turret.dm]: typecast this as a turf

File: code/game/machinery/porta_turret/portable_turret.dm
Comment (RemieRichards) [inline code/game/machinery/porta_turret/portable_turret.dm]: Should delete itself if it doesn't get a turret (eg: admin spawning these abstract-dontspawnpls-hand objects)

File: code/game/machinery/porta_turret/portable_turret.dm
Comment (RemieRichards) [inline code/game/machinery/porta_turret/portable_turret.dm]: no ```var/``` in proc arguments

check the rest of your PR.

File: code/game/machinery/porta_turret/portable_turret.dm
Comment (RemieRichards) [inline code/game/machinery/porta_turret/portable_turret.dm]: why do you pass ```target_turf``` as an arg? it's already on the gun...

**Comment by RemieRichards** [comment]
I want my review addressed, and I'd like to see this tested.

**Comment by Robustin** [comment]
I love the dialogue between Steelpoint and LeoZ, classic argument with LeoZ.

"I dont like because [strawman]"
"But there's no [strawman]"
"Yea I still dont like it"

File: code/game/machinery/porta_turret/portable_turret.dm
Comment (Robustin) [inline code/game/machinery/porta_turret/portable_turret.dm]: been doing this as a precaution since I've seen it all over, thanks for clarifying

File: code/game/machinery/porta_turret/portable_turret.dm
Comment (Robustin) [inline code/game/machinery/porta_turret/portable_turret.dm]: I think this proc used to be in the controls and didn't have the var or something

**Comment by lzimann** [comment]
At least it's not an essay telling the story of your life that no one bothers to read.

**Comment by Robustin** [comment]
Raviolio Ravioli give me the mergioli

**Pull request merged**
